File: src/BjjEire.Domain/Entities/Gyms/Gym.cs
// Copyright (c) BjjEire. All rights reserved.
// Licensed under the MIT License.

using BjjEire.Domain.Entities.Common;
using BjjEire.Domain.Enums;

namespace BjjEire.Domain.Entities.Gyms;

public class Gym : BaseEntity
{
    [BsonElement("name")]
    public string Name { get; set; } = string.Empty;

    [BsonElement("description")]
    public string? Description { get; set; } = string.Empty;

    [BsonElement("status")]
    public GymStatus Status { get; set; }

    [BsonElement("county")]
    public County County { get; set; }

    [BsonElement("affiliation")]
    public Affiliation? Affiliation { get; set; } = new();

    [BsonElement("trialOffer")]
    public TrialOffer TrialOffer { get; set; } = new();

    [BsonElement("location")]
    public Location Location { get; set; } = new();

    [BsonElement("socialMedia")]
    public SocialMedia SocialMedia { get; set; } = new();

    [BsonElement("offeredClasses")]
    public List<ClassCategory> OfferedClasses { get; set; } = [];

    [BsonElement("website")]
    public string? Website { get; set; }

    [BsonElement("timetableUrl")]
    public string? TimetableUrl { get; set; }

    [BsonElement("imageUrl")]
    public string? ImageUrl { get; set; }
}

File: src/bjjeire-api/src/main/java/com/bjjeire/api/gym/Gym.java
package com.bjjeire.api.gym;

import com.bjjeire.api.common.County;
import com.bjjeire.api.common.Location;
import com.bjjeire.api.common.SocialMedia;
import java.time.Instant;
import java.util.ArrayList;
import java.util.List;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;
import org.springframework.data.mongodb.core.mapping.Field;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Document(Gym.ENTITY_NAME)
public class Gym {
    public static final String ENTITY_NAME = "Gym";

    @Id
    private String id;

    private String name;
    private String description;
    private GymStatus status;
    private County county;
    private Affiliation affiliation;
    private TrialOffer trialOffer;
    private Location location;
    private SocialMedia socialMedia;

    @Builder.Default
    private List<ClassCategory> offeredClasses = new ArrayList<>();

    private String website;
    private String timetableUrl;
    private String imageUrl;

    @Field("createdAt")
    private Instant createdOnUtc;

    @Field("createdBy")
    private String createdBy;

    @Field("updatedAt")
    private Instant updatedOnUtc;

    @Field("updatedBy")
    private String updatedBy;
}
